src/components/views/auth/CountryDropdown.tsx

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2017 Vector Creations Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import * as sdk from '../../../index';
+
+import {COUNTRIES, getEmojiFlag} from '../../../phonenumber';
+import SdkConfig from "../../../SdkConfig";
+import { _t } from "../../../languageHandler";
+
+const COUNTRIES_BY_ISO2: Record<string, Country> = {};
+
+for (const c of COUNTRIES) {
+    COUNTRIES_BY_ISO2[c.iso2] = c;
+}
+
+interface Country {
+    iso2: string;
+    name: string;
+    prefix: string;
+}
+
+interface IProps {
+    className?: string;
+    showPrefix?: boolean;
+    onOptionChange: (country: Country) => void;
+
+    /** The country code to use. */
+    value?: string;
+
+    /** Whether or not to append a numeral prefix to the value. */
+    isSmall?: boolean;
+
+    disabled?: true;
+}
+
+function countryMatchesSearchQuery(query, country) {
```

Comment:
```suggestion
function countryMatchesSearchQuery(query: string, country: Country) {
```

Add parameter types here.